cmd/kops/create_cluster.go

```diff
@@ -372,22 +378,36 @@ func RunCreateCluster(f *util.Factory, cmd *cobra.Command, args []string, out io
 		}
 	}
 
+	//Bastion
+	if c.Topology == api.TopologyPublic && c.Bastion == true {
+		return fmt.Errorf("Bastion supports --topology='private' only.")
+	}
+
 	// Network Topology
 	switch c.Topology {
 	case api.TopologyPublic:
-		cluster.Spec.Topology = &api.TopologySpec{Masters: api.TopologyPublic, Nodes: api.TopologyPublic, BypassBastion: false}
+		cluster.Spec.Topology = &api.TopologySpec{Masters: api.TopologyPublic, Nodes: api.TopologyPublic}
+		cluster.Spec.Bastion = &api.BastionSpec{Enable: c.Bastion}
```

Comment:
The bastion is a concept of private topology, not the cluster.. Can we get the `&BastionSpec{}` into `cluster.Spec.Topology.Bastion` instead?